File: Cplusplus/Algorithms/bubbleSort.cpp

int* bubbleSort(int arr[], int sizeArr)
{
	int temp;
	bool swapped = true;
while (swapped)
{
swapped = false;
	for (int i=0;i<sizeArr-1; i++)
	{	
		if (arr[i] > arr[i+1])
		{
			swapped = true;
			temp = arr[i];
			arr[i] = arr[i+1];
			arr[i+1] = temp;	
		}	 
	}
}
return arr;
}

File: Matlab/Algorithms/bubbleSort.m
function [array] = bubbleSort(array)
%% This function sorts a given array by bubble sort. 

swapped = true;

while swapped
    swapped = false;
    for i = 1:length(array)-1
        if array(i) > array(i+1)
            swapped = true;
           temp = array(i);
           array(i) = array(i+1);
           array(i+1) = temp;
        end
    end
end


end